addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Passive WebSocket Scan Template.js

```diff
@@ -0,0 +1,87 @@
+ * helper.getAlertRaiser() -> Returns an WebSocketAlertRaiser instance which is used
+ *                            for building and raising alerts.
+
+ * * Some useful functions about WebSocketAlertRaiser:
+ * * alertRaiser.setRiskConfidence(risk, confidence) -> Sets the Risk and the Confidence of the alert. (by default RISK_INFO, CONFIDENCE_MEDIUM).
+ * * alertRaiser.setName(name)                       -> Sets the name (by default "").
+ * * alertRaisersetDescription(description)          -> Sets a description about potential threat (by default "").
+ * * alertRaiser.setParam(param)                     -> Sets in which parameter the threat noticed (by default "").
+ * * alertRaiser.setAttack(attack)                   -> Sets the possible attack (by default "").
+ * * alertRaiser.setMessage(msg)                     -> Sets the WebSocketMessageDTO.
+ * * alertRaiser.setSolution(solution)               -> Sets a possible solution (by default "").
+ * * alertRaiser.setReference(reference)             -> Sets extra references (ex. a web link) (by default "").
+ * * alertRaiser.setEvidence(evidence)               -> Sets what's the evidence of the potential thread (by default "").
+ * * alertRaiser.setCweIdm(cweId)                    -> Sets the CWE ID of the issue (by default 0)
+ * * alertRaiser.setWascId(wascId)                   -> Sets the WASC ID of the issue (by default 0)
+ * * alertRaiser.raise()                             -> Build and Raise the alert (returns the WebSocketAlertWrapper)
+
+ * @param msg - the Websocket Message being scanned. This is a WebSocketMessage object.
+ *
+ * Some useful functions and fields of WebSocketMessageDTO:
+ * msg.channel        -> Channel of the message (WebSocketChannelDTO)
+ * msg.id             -> Unique ID of the message (int)
+ * msg.opcode         -> Opcode of the message (int) (Opcodes defined bellow)
+ * msg.readableOpcode -> Textual representation of opcode (String)
+ * msg.isOutgoing     -> Outgoing or incoming message (boolean)
+ * msg.getReadablePayload() -> Return readable representation of payload
+ *
+ * * Some useful functions and fields of WebSocketChannelDTO:
+ * * channel.id         -> Unique ID of the message (int)
+ * * channel.host       -> Host of the WebSocket Server (String)
+ * * channel.port       -> Port where the channel is connected at. Usually at 80 or 443.
+ * * channel.url        -> URL used in HTTP handshake.
+ */
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    // Test the request or response here
+    print(msg.getReadablePayload());
+
+    if(true){
+        helper.getAlertRaiser()
+            .setRiskConfidence(RISK_INFO, CONFIDENCE_LOW)
+            .setName("Name of the alert")
+            .setDescription("Description of the Alert.")
+            .setParam("Parameter of the Alert.")
+            .setAttack("Attack of the Alert.")
+            .setMessage("Message of the Alert.")
```

Comment:
This should be removed, note that this leads to an exception (this would have to be the WebSocket message not a string).

Reply:
Fixed in 17ea0de3a